### PR TITLE
Delete Reminder and display reminder

### PR DESCRIPTION
**Pull request by hshiah**
I will modify the parser part after all works done.

### CODE REVIEW

File: src/main/java/seedu/financialplanner/commands/DeleteReminderCommand.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package seedu.financialplanner.commands;
+import seedu.financialplanner.reminder.ReminderList;
+import seedu.financialplanner.utils.Ui;
+
+public class DeleteReminderCommand extends Command{
+    private final int index;
+
+    public DeleteReminderCommand(RawCommand rawCommand) throws IllegalArgumentException {
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/commands/DeleteReminderCommand.java:11]: possible to add logging in this constructor

File: src/main/java/seedu/financialplanner/commands/DeleteReminderCommand.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        if(!rawCommand.extraArgs.isEmpty()){
+            String unknownExtraArgument = new java.util.ArrayList<>(rawCommand.extraArgs.keySet()).get(0);
+            throw new IllegalArgumentException(String.format("Unknown extra argument: %s", unknownExtraArgument));
+        }
+    }
+
+    @Override
+    public void execute() {
+        ReminderList.getInstance().deleteReminder(index);
+        Ui.getInstance().showMessage("You have deleted "+ReminderList.getInstance().list.get(index-1));
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/commands/DeleteReminderCommand.java]:
```suggestion
        Ui.getInstance().showMessage("You have deleted " + ReminderList.getInstance().list.get(index-1));
```

File: src/main/java/seedu/financialplanner/commands/DeleteReminderCommand.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+            index = Integer.parseInt(stringIndex);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Index must be an integer");
+        }
+
+        if (index == 0) {
+            throw new IllegalArgumentException("Index must be within the list");
+        }
+
+        if( index > ReminderList.getInstance().list.size()+1){
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/commands/DeleteReminderCommand.java]:
```suggestion
        if (index > ReminderList.getInstance().list.size()+1){
```

File: src/main/java/seedu/financialplanner/commands/MarkReminderCommand.java [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+        try {
+            index = Integer.parseInt(stringIndex);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Index must be an integer");
+        }
+        if (index == 0) {
+            throw new IllegalArgumentException("Index must be within the list");
+        }
+        if( index > ReminderList.getInstance().list.size()+1){
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/commands/MarkReminderCommand.java]:
```suggestion
        if (index > ReminderList.getInstance().list.size()+1){
```

File: src/main/java/seedu/financialplanner/commands/MarkReminderCommand.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
+package seedu.financialplanner.commands;
+import seedu.financialplanner.reminder.ReminderList;
+import seedu.financialplanner.utils.Ui;
+public class MarkReminderCommand extends Command{
+    private final int index;
+    public MarkReminderCommand(RawCommand rawCommand) throws IllegalArgumentException {
+        String stringIndex;
+        if(rawCommand.args.size() == 1) {
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/commands/MarkReminderCommand.java:9]:
```suggestion
        if (rawCommand.args.size() == 1) {
```

File: src/main/java/seedu/financialplanner/commands/MarkReminderCommand.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
+package seedu.financialplanner.commands;
+import seedu.financialplanner.reminder.ReminderList;
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/commands/MarkReminderCommand.java:3]: should there be spacing between package and imports

File: src/main/java/seedu/financialplanner/commands/MarkReminderCommand.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        rawCommand.extraArgs.remove("i");
+        if(!rawCommand.extraArgs.isEmpty()){
+            String unknownExtraArgument = new java.util.ArrayList<>(rawCommand.extraArgs.keySet()).get(0);
+            throw new IllegalArgumentException(String.format("Unknown extra argument: %s", unknownExtraArgument));
+        }
+    }
+
+    @Override
+    public void execute() {
+        ReminderList.getInstance().list.get(index-1).markAsDone();
```
Comment (NeoMinWei) [inline src/main/java/seedu/financialplanner/commands/MarkReminderCommand.java:35]: maybe can make reminderlist and ui members in the class so don't have to keep using getInstance

**Review by NeoMinWei — APPROVED** [review]
LGTM, how do u want to change the parser?

**Comment by hshiah** [comment]
> LGTM, how do u want to change the parser?

I haven't added the command to parser to avoid conflict.
After all features implemented, I will add these commands to parser.

**Comment by NeoMinWei** [comment]
> > LGTM, how do u want to change the parser?
> 
> I haven't added the command to parser to avoid conflict. After all features implemented, I will add these commands to parser.

ok

File: src/main/java/seedu/financialplanner/commands/ReminderListCommand.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+    public ReminderListCommand(RawCommand rawCommand) throws IllegalArgumentException{
+
+    }
+
+    @Override
+    public void execute() {
+        Ui ui = Ui.getInstance();
+        ReminderList reminderList = ReminderList.getInstance();
+        ui.showMessage("Here is your reminder list:");
+        ui.showMessage(ReminderList.getInstance().list.toString());
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/commands/ReminderListCommand.java]:
```suggestion
        ui.showMessage(reminderList.list.toString());
```
No need another getInstance() here i think

File: src/main/java/seedu/financialplanner/commands/UnmarkReminderCommand.java [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+
+        try {
+            index = Integer.parseInt(stringIndex);
+        } catch (IllegalArgumentException e) {
+            throw new IllegalArgumentException("Index must be an integer");
+        }
+        if (index == 0) {
+            throw new IllegalArgumentException("Index must be within the list");
+        }
+        if( index > ReminderList.getInstance().list.size()+1){
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/commands/UnmarkReminderCommand.java]:
```suggestion
        if (index > ReminderList.getInstance().list.size()+1){
```

File: src/main/java/seedu/financialplanner/commands/UnmarkReminderCommand.java [hunk 10]
```diff
@@ -0,0 +1,37 @@
+package seedu.financialplanner.commands;
+import seedu.financialplanner.reminder.ReminderList;
+import seedu.financialplanner.utils.Ui;
+public class UnmarkReminderCommand extends Command{
+    private final int index;
+    public UnmarkReminderCommand(RawCommand rawCommand) throws IllegalArgumentException {
+        String stringIndex;
+        if(rawCommand.args.size() == 1) {
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/commands/UnmarkReminderCommand.java]:
```suggestion
        if (rawCommand.args.size() == 1) {
```

File: src/main/java/seedu/financialplanner/goal/Goal.java [hunk 11]
```diff
@@ -12,15 +12,17 @@ public Goal(String label, int amount) {
 
     public String toString() {
         String status = isDone ? "Done" : "Not Done";
-        return "Goal: " + this.label + " | " + this.amount + " | " + status;
+        return "Goal " + System.lineSeparator()+ "   Label: " + label + System.lineSeparator() + "   Amount: " +
```
Comment (ryan1604) [inline src/main/java/seedu/financialplanner/goal/Goal.java:15]: Should there be a ":" after Goal? Same thing for reminder

**Review by ryan1604 — APPROVED** [review]
LGTM, other than a few minor issues like standardising the spacing between methods, lines and package/imports to make the code more readable:D

File: src/main/java/seedu/financialplanner/goal/Goal.java (hunk 11 above)
Comment (hshiah) [reply src/main/java/seedu/financialplanner/goal/Goal.java:15]: This part I follow the expense, which is without a ":".

**Pull request merged**